### ISSUE TITLE
CSV uploads - allow selecting the number format for uploads independently of the display number format

### ISSUE THREAD

**Issue by zbodi74** [metabase/metabase#45171]
**Is your feature request related to a problem? Please describe.**
Currently the Separator Style setting in Admin Settings / Localization / Numbers / Separator Style affects how numbers in uploaded files are parsed. This may occasionally lead to unexpected behavior.

For example, setting the display number format to `100.000,00` (used in Denmark) causes the parser to ignore the decimal comma, and a number like `18500.00` will be parsed as the integer `1850000`.

**Describe the solution you'd like**
Ideally, the number format for parsing CSV files should be configurable separately from the display number format. 
This would allow users to set the number format for their CSV that might differ from the one preferred to displaying them.

**Describe alternatives you've considered**
Currently, there are no good alternatives. Users must use the same number format for both display and CSV parsing. If their input format differs from their display preference, they either need to convert the file or select a less desirable display format compatible with their input.

**How important is this feature to you?**
Reported by a customer - I used their actual example above.

**Additional context**
n/a

**Comment by ds-clearago** [comment]
I agree – effectively Metabase is the only tool now where we have to convert `.`-decimals to `,`-decimals prior to uploading into the system just because our display settings are set to Germany (where our clients reside).

Everywhere else where we export/import CSVs the values are comma-separated while no thousands separator is used and decimals are split by the international digital `.`

Since almost every other code denotes floats with `.` just metabase CSV doesn't that's very unintuitive behavior – and has already caused us multiple wrong dashboards for clients as there was a mixup between usage of `.` and `,`... Can I upvote this somehow?
